confirmation/models.py

```diff
@@ -139,6 +139,19 @@ def send_confirmation(self, obj, email_address, additional_context=None,
         send_mail(subject, body, settings.DEFAULT_FROM_EMAIL, [email_address], html_message=html_content)
         return self.create(content_object=obj, date_sent=now(), confirmation_key=confirmation_key)
 
+class EmailChangeConfirmationManager(ConfirmationManager):
+    def get_activation_url(self, key, host=None):
+        # type: (Text, Optional[str]) -> Text
+        if host is None:
+            host = settings.EXTERNAL_HOST
+        return u'%s%s%s' % (settings.EXTERNAL_URI_SCHEME,
+                            host,
+                            reverse('zerver.views.user_settings.confirm_email_change',
+                                    kwargs={'confirmation_key': key}))
+
+    def get_link_validity_in_days(self):
+        # type: () -> int
+        return getattr(settings, 'EMAIL_CHANGE_CONFIRMATION_DAYS', 1)
```

Comment:
Just put `EMAIL_CHANGE_CONFIRMATION_DAYS: 1,` in DEFAULT_SETTINGS and make this an explicit access.  It'll do the same thing, but `DEFAULT_SETTINGS` should have every user-controllable setting in it so they're easy to find.